kani-compiler/src/codegen_boogie/context/kani_intrinsic.rs

```diff
@@ -26,23 +26,25 @@ pub enum KaniIntrinsic {
     KaniAssume,
 }
 
-impl<'tcx> BoogieCtx<'tcx> {
-    /// If provided function is a Kani intrinsic (e.g. assert, assume, cover), returns it
-    // TODO: move this function up to `kani_middle` along with the enum
-    pub fn kani_intrinsic(&self, instance: Instance<'tcx>) -> Option<KaniIntrinsic> {
-        let intrinsics = KaniIntrinsic::VARIANTS;
-        for intrinsic in intrinsics {
-            let attr_sym = rustc_span::symbol::Symbol::intern(intrinsic);
-            if let Some(attr_id) = self.tcx.all_diagnostic_items(()).name_to_id.get(&attr_sym) {
-                if instance.def.def_id() == *attr_id {
-                    debug!("matched: {:?} {:?}", attr_id, attr_sym);
-                    return Some(KaniIntrinsic::from_str(intrinsic).unwrap());
-                }
+/// If provided function is a Kani intrinsic (e.g. assert, assume, cover), returns it
+// TODO: move this function up to `kani_middle` along with the enum
```

Comment:
+1

Reply:
I'll try to do that as part of merging the Boogie backend to main.